### ISSUE TITLE
Feature - API support

### ISSUE THREAD

**Issue by Regala** [yogeshojha/rengine#181]
**Is your feature request related to a problem? Please describe.**
Support a simple API which we can use to integrate with other tools.

**Describe the solution you'd like**
e.g. I'd like to programmatically push results of my own to the platform, say a new domain found or the result of a new content discovery. Something like `POST /target/yahoo/ {...}`.  

**Additional context**
Ideally we could also query some of the data.

**Comment by VaibhavNagariya** [comment]
The ability to push the target file using API would be a great add-on too.

**Comment by Regala** [comment]
ReNgine v2?

**Comment by yogeshojha** [comment]
Nope, should be out by this month's end. The APIs exist, but the lack of documentation makes it difficult for anyone to use. I have started the documentation for API, so should not take very long.

**Comment by yogeshojha** [comment]
Probably v1.1? ;) Documentation along with a few important features are lined up.

1.1 will come with an option to scan individual subdomains for further scan, suppose, initially you do only the subdomain scan, and later, you think redacted.example.com may be good to go for Directory scan, you can choose to do so from subdomains list table, from the action menu, you can trigger the further scans. Like right now, everything goes into directory scan, vulnerability scan etc, you can choose per say subdomains.

**Comment by serhattsnmz** [comment]
Hi, is there any news about APIs or v1.1?

**Comment by RoiSec** [comment]
+1

**Comment by RoiSec** [comment]
@yogeshojha ?

**Comment by Regala** [comment]
Hi @yogeshojha , any chance this is in the near future plans? 🥺

**Comment by sneakyrussianee** [comment]
Hello @yogeshojha  any info on the API documentation? Thanks

**Comment by d4n5h** [comment]
@yogeshojha Any estimation of when the API docs will be available?

**Comment by psyray** [comment]
@yogeshojha Could you push the doc if you have it ?

**Comment by AnonymousWP** [comment]
@yogeshojha What do we do with this open issue?

**Comment by psyray** [comment]
We could link this issue with the new tool created
https://github.com/glownd/rengine-tool/

**Comment by glownd** [comment]
@psyray @yogeshojha @AnonymousWP -- you're more than welcome to integrate it if you'd like.

**Comment by AnonymousWP** [comment]
I don't really have an opinion about this. Let's wait for Yogesh.

**Comment by ipsecc** [comment]
Any updates on it?

**Comment by twf0** [comment]
Update
